cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/ConstraintCheckerService.java

```diff
@@ -308,5 +322,36 @@ private ConstraintResult.SatisfiedState constraintsSatisfied(Job job, long now)
       return satisfiedState;
     }
 
+    private void emitScheduleJobSuccessAndLatencyMetric(String application, String schedule, long jobCreationTime) {
+      if (metricsCollectionService == null) {
+        return;
```

Comment:
> Also given that metricsCollectionService is being injected, it should never be null (unless provider returns null).

yes, since, we can't guarantee provider wouldn't return null, we should ideally do the null check right?

Reply:
It's a bad practice for making provider to return null. Normally either we have a binding and correct object gets injected, or there is no binding, and we get an injection error. Because of that, we normally do not check for null in cases where objects are being injected. 
In cases where arguments can be null, we should annotate it so to avoid confusion.

Reply:
> It's a bad practice for making provider to return null. Normally either we have a binding and correct object gets injected, or there is no binding, and we get an injection error. Because of that, we normally do not check for null in cases where objects are being injected. In cases where arguments can be null, we should annotate it so to avoid confusion.

understood.